reference/debugging.md

```diff
@@ -0,0 +1,284 @@
+
+If the generated Envoy configuration does not look correct, you can edit and restart the Envoy process. We do not recommend doing this in production.
+
+1. Scale down the Ambassador Deployment to a single pod in order to send all Ambassador traffic through a single instance. Use the following command: `kubectl scale deployment ambassador --replicas=1`
+2. Wait for the scale down to complete, and then use the exec command to access the Pod: `kubectl exec -it ambassador-85c4cf67b-4pfj2 -- /bin/sh`
+3. Modify the `envoy/envoy.json` file: `/ambassador $ vi envoy/envoy.json`
+4. Get the PID of the `ambex` process with: `ps -ef | grep ambex`
+5. Then, restart the `ambex` process which will pass the updated `envoy.json` to Envoy with: `kill -HUP $AMBEX_PID`
+6. In your Edge Policy Console, go to **Diagnostics > Logging** and choose the “set log level to debug” option.
+7. Verify that the restart and new configuration was successful with the correct configuration by following the Ambassador/Envoy logs.
+
+Be aware that even though you have modified the configuration files, the Edge Policy Console and/or the Diagnostics service may not accurately reflect your updates.
+
+The command history will look similar to the following:
+
+```console
+$ kubectl scale deployment ambassador --replicas=1
+deployment.apps/ambassador scaled
+tmp $ kubectl get pods
+NAME                         READY     STATUS        RESTARTS   AGE
+ambassador-85c4cf67b-4pfj2   1/1       Running       0          30m
+ambassador-85c4cf67b-fqp9g   1/1       Terminating   0          30m
+ambassador-85c4cf67b-vg6p5   1/1       Terminating   0          30m
+
+$ kubectl exec -it ambassador-85c4cf67b-4pfj2 -- /bin/sh
+/ambassador $ ls -lsa
+total 64
+    8 drwxrwxr-x    1 root     root          4096 Apr  5 21:01 .
+    4 drwxr-xr-x    1 root     root          4096 Apr  5 21:01 ..
+    4 drwxr-xr-x    3 8888     root          4096 Apr  5 21:01 .cache
+    4 drwxrwxr-x    1 root     root          4096 Apr  5 17:38 ambassador-config
+    4 drwxrwxr-x    1 root     root          4096 Apr  5 17:38 ambassador-demo-config
+    4 -rw-r--r--    1 8888     root             2 Apr  5 21:01 ambex.pid
+    4 -rw-r--r--    1 8888     root          1626 Apr  5 21:01 bootstrap-ads.json
+    4 drwxrwxr-x    1 root     root          4096 Apr  5 17:38 demo-services
+    8 -rwxr-xr-x    1 root     root          6589 Apr  5 17:37 entrypoint.sh
+    4 drwxrwxr-x    1 root     root          4096 Apr  5 21:01 envoy
+    4 -rwxr-xr-x    1 root     root           584 Apr  5 17:37 kick_ads.sh
+    4 -rwxr-xr-x    1 root     root          4007 Apr  5 17:37 kubewatch.py
+    4 -rwxr-xr-x    1 root     root           484 Apr  5 17:37 post_update.py
+    4 drwxr-xr-x    3 8888     root          4096 Apr  5 21:01 snapshots
+/ambassador $ vi envoy/envoy.json
+/ambassador $ ps -ef | grep ambex
+  33 8888     21:56 ambex /ambassador/envoy
+ 122 8888      0:00 grep ambex
+/ambassador $ kill -HUP 33
+```
```

Comment:
This whole section about updating the Envoy config is flat wrong. It is _possible_ to do it by modifying `$AMBASSADOR_CONFIG_BASE_DIR/envoy/envoy.json` and sending `ambex` a `HUP` as described above, but it's a _terrible_ idea because it'll get overwritten at the drop of a hat. Let's just strike the whole section.